src/print.jl

```diff
@@ -147,34 +147,92 @@ wrap_in_inline_math_mode(str) = "\$ $str \$"
 ## Model
 #------------------------------------------------------------------------
 function Base.show(io::IO, model::Model)
+    plural(n) = (n==1 ? "" : "s")
     println(io, "A JuMP Model")
+    sense = objective_sense(model)
+    if sense == MOI.MaxSense
+        print(io, "Maximization")
+    elseif sense == MOI.MinSense
+        print(io, "Minimization")
+    else
+        print(io, "Feasibility")
+    end
+    println(io, " problem with:")
     # TODO: Consider allowing a JuMP model to have a string name.
-    println(io, "Variables: ", num_variables(model))
+    println(io, "Variable", plural(num_variables(model)), ": ",
+            num_variables(model))
     # https://github.com/JuliaOpt/JuMP.jl/issues/1556
     # TODO: This doesn't account for nonlinear objectives
     # println(io, "\tObjective function type:",
     #            MOI.get(model, MOI.ObjectiveFunctionType()))
     constraint_types = MOI.get(model, MOI.ListOfConstraints())
     for (F, S) in MOI.get(model, MOI.ListOfConstraints())
         num_constraints = MOI.get(model, MOI.NumberOfConstraints{F, S}())
-        println(io, "`$F`-in-`$S`: $num_constraints constraints")
+        println(io, "`$F`-in-`$S`: $num_constraints constraint",
+                plural(num_constraints))
     end
     if !iszero(num_nl_constraints(model))
-        println(io, "Nonlinear: ", num_nl_constraints(model), " constraints")
+        println(io, "Nonlinear: ", num_nl_constraints(model), " constraint",
+                plural(num_nl_constraints(model)))
     end
     model_mode = mode(model)
     println(io, "Model mode: ", model_mode)
     if model_mode == Manual || model_mode == Automatic
         println(io, "CachingOptimizer state: ",
                 MOIU.state(backend(model)))
     end
-    println(io, "Solver name: ", solver_name(model))
+    # The last print shouldn't have a new line
+    print(io, "Solver name: ", solver_name(model))
     names_in_scope = collect(keys(object_dictionary(model)))
     if !isempty(names_in_scope)
-        println(io, "Names registered in the model: ",
-                    join(string.(names_in_scope), ", "))
+        println(io)
+        print(io, "Names registered in the model: ",
+              join(string.(names_in_scope), ", "))
+    end
+end
+
+function Base.print(io::IO, model::Model)
+    print(io, model_string(REPLMode, model))
+end
+function Base.show(io::IO, ::MIME"text/latex", model::Model)
+    print(io, wrap_in_math_mode(model_string(IJuliaMode, model)))
+end
+function model_string(print_mode, model::Model)
+    ijl = print_mode == IJuliaMode
```

Comment:
We could consider dropping IJuliaMode as @IainNZ suggested. Doesn't need to be done now.

Reply:
I am not convinced we should drop it